src/ai/ai_gui.cpp

```diff
@@ -642,17 +643,32 @@ struct ScriptTextfileWindow : public TextfileWindow {
 
 	ScriptTextfileWindow(TextfileType file_type, CompanyID slot) : TextfileWindow(file_type), slot(slot)
 	{
-		const char *textfile = GetConfig(slot)->GetTextfile(file_type, slot);
-		this->LoadTextfile(textfile, (slot == OWNER_DEITY) ? GAME_DIR : AI_DIR);
+		this->OnInvalidateData();
 	}
 
 	void SetStringParameters(int widget) const override
 	{
 		if (widget == WID_TF_CAPTION) {
 			SetDParam(0, (slot == OWNER_DEITY) ? STR_CONTENT_TYPE_GAME_SCRIPT : STR_CONTENT_TYPE_AI);
-			SetDParamStr(1, GetConfig(slot)->GetName());
+
+			if (GetConfig(slot)->GetInfo() != NULL) {
+				SetDParamStr(1, GetConfig(slot)->GetInfo()->GetName());
+			} else {
+				char name[1024];
+				GetString(name, (slot == OWNER_DEITY) ? STR_AI_CONFIG_NONE : STR_AI_CONFIG_RANDOM_AI, lastof(name));
+				const char *script_name = stredup(name);
+				SetDParamStr(1, script_name);
+			}
 		}
 	}
+
+	void OnInvalidateData(int data = 0, bool gui_scope = true) override
+	{
+		const char *textfile = GetConfig(slot)->GetTextfile(file_type, slot);
+		if (textfile == NULL) delete this;
```

Comment:
No clue what you are smoking here exactly, but this is clearly wrong. Deleting `this` and using it afterwards. That is .. yeah. Clearly wrong.

Reply:
Thx, that was it.